shell/machine-config/azure.vue

```diff
@@ -177,19 +199,192 @@ export default {
     },
   },
 
+  computed: {
+    locationOptionsInDropdown() {
+      const locationOptionsCopy = [...this.locationOptions];
+
+      return locationOptionsCopy.sort((a, b) => {
+        // Hopefully it's easier to find a region if the list is in
+        // alphabetical order.
+        if (a.name > b.name) {
+          return 1;
+        }
+        if (a.name < b.name) {
+          return -1;
+        }
+
+        return 0;
+      } )
+        .map((option) => {
+          return {
+            displayName: `${ option.displayName } (${ option.name })`,
+            name:        option.name,
+          };
+        });
+    },
+    vmsWithAcceleratedNetworking() {
+      return this.vmSizes.filter((vmData) => {
+        return vmData.AcceleratedNetworkingSupported;
+      });
+    },
+    vmsWithoutAcceleratedNetworking() {
+      return this.vmSizes.filter((vmData) => {
+        return !vmData.AcceleratedNetworkingSupported;
+      });
+    },
+    selectedVmSizeSupportsAN() {
+      const selectedSizeIsValid = !!this.vmsWithAcceleratedNetworking.find((vmData) => {
+        return this.value.size === vmData.Name;
+      });
+
+      return selectedSizeIsValid;
+    },
+    vmSizeAcceleratedNetworkingWarning() {
+      if (!this.selectedVmSizeSupportsAN) {
```

Comment:
I am seeing the accelerated networking warning even when I haven't checked that box. I think this needs a `&& this.value.acceleratedNetworking`

Reply:
I think that's left over from when the checkbox was disabled to explain why it was disabled. But then I took out the disabled property to prevent the weird state where the checkbox could be both checked and disabled in edit mode. I've updated it so that the error appears when the box is checked but it's not supported